### PR TITLE
Replace Azure Pipelines with Gitlab CI

### PR DESCRIPTION
**Pull request by esteve**
Removed Azure Pipelines and added Gitlab CI in preparation for the Gitlab migration.

---
This change is [Reviewable](https://reviewable.io/reviews/apexai/performance_test/96)

### CODE REVIEW

**Comment by esteve** [comment]
@daggarwa I've readded Gitlab CI and also added a new job for cross compiling `performance_test`

**Comment by daggarwa** [comment]
> @daggarwa I've readded Gitlab CI and also added a new job for cross compiling `performance_test`

@esteve Just to clarify. From the changes you mentioned , now we have the ability to build and run ApexOS dependent stuff too in this repository?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,69 +0,0 @@
...
-        git clone https://github.com/eclipse-cyclonedds/cyclonedds.git
-        cd cyclonedds
-        git checkout 4e805597631ed0dcbdc0eecfe9d532cb75180ae7 # Head as of 20190725
-        mkdir -p build
-        cd build
-        cmake -DCMAKE_INSTALL_PREFIX=/opt/cyclonedds ../src
-        cmake --build .
-        sudo cmake --build . --target install
-        find /opt/cyclonedds
-      displayName: Build and install Cyclone DDS
```
Comment (daggarwa) [inline azure-pipelines.yml]: @esteve Can you please explain why is this removed in the gitlab CI script?

**Comment by daggarwa** [comment]
@esteve Will this fail https://github.com/ApexAI/performance_test/pull/90 if we merge in the Gitlab CI since its not currently containing any job to install `Cyclone DDS` whereas `Azure` did?

File: azure-pipelines.yml (hunk 1 above)
Comment (esteve) [reply azure-pipelines.yml]: @daggarwa I forgot about Cyclone, I'll add it to Gitlab CI

Comment (daggarwa) [reply azure-pipelines.yml]: @esteve Sounds good. Thanks!

**Comment by daggarwa** [comment]
@esteve The CI pipeline is failing currently due to some `Mavin` dependency.

**Comment by daggarwa** [comment]
@esteve The CI pipeline is failing now due to `$ cp LICENSE /opt/performance_test
cp: cannot stat 'LICENSE': No such file or directory` . I am not sure what is this error about.

**Comment by esteve** [comment]
@daggarwa Thanks. I'm working on it, but didn't have time to finish it yesterday.

**Comment by esteve** [comment]
@daggarwa I've fixed the last issue and the Gitlab CI now passes. I can't remove the Azure trigger until this PR has been merged, so it's fine the second check is still on hold.

File: azure-pipelines.yml (hunk 1 above)
Comment (esteve) [reply azure-pipelines.yml]: Fixed in 90cd76f

**Comment by jpsamper2009** [comment]
Since we can't clear the status above (which refers to the old azure pipeline), I'll merge this manually.

**Comment by esteve** [comment]
@jpsamper2009 thanks!

@daggarwa we'll have to update the other PRs to use the new CI

**Pull request merged**